File: lua/wordcount.lua
#!/usr/bin/env lua

local counts = setmetatable({}, {__index=function() return 0 end})
for line in io.stdin:lines() do
	for word in line:gmatch("%S+") do
		counts[word] = counts[word] + 1
	end
end
local sorted = {}
local n = 0
for word in pairs(counts) do
	n = n + 1
	sorted[n] = word
end
table.sort(sorted, function(a,b)
	local na = counts[a]
	local nb = counts[b]
	if na == nb then
		return a < b
	else
		return na > nb
	end
end)
for i=1, n do
	local word = sorted[i]
	print(word, counts[word])
end

File: python3/wordcount.py
#!/usr/bin/env python3
from collections import defaultdict
from sys import stdin
import codecs

stdin = codecs.getreader('utf8')(stdin.detach(), errors='ignore')


def word_count():
    counter = defaultdict(int)
    for l in stdin:
        for word in bytes(l, 'utf8').split():
            counter[word] += 1
    for word, cnt in sorted(counter.items(), key=lambda x: (-x[1], x[0])):
        print('{0}\t{1}'.format(word.decode('utf8'), cnt))

if __name__ == '__main__':
    word_count()
